src/sysen3/papsav.1

```diff
@@ -0,0 +1,372 @@
+;;; MAIN PROGRAM
+
+i==4
+l==5
+v==6
+lim==7
+remapt==2000
+
+    loc 100
+.entr.: movei tt,t1(p)
+        hrli tt,.prsva
+        add p,.retn.+7
+        blt tt,(p)
+        move .ap.,t1
+        move .pf.,.cf.
+        move .cf.,p
+        add p,(t)
+        jumpl p,t1(t)
+        .suset .retn.+11
+        jrst t1(t)
+.retn.: move p,.cf.
+        move .cf.,.pf.
+        movsi tt,-11(p)
+        hrri tt,.prsva
+        blt tt,.pf.
+        sub p,.retn.+7
+        popj p,
+        count,,count
+        200000
+        setz .retn.+10(.ap.)
+        0
+fmtbl:  0
+        typoct ;1 full word octal
+        typdec ;2 full wd decimal
+        typhaf ;3 " with commas
+        tycrlf ;4 do cr
+        cpopj  ;5 ?
+        typsix ;6 sixbit
+        typasz ;7 asciz
+
+cpopj:  popj p,
+
+feep:   blt b,(.prsva)
+crlf:   asciz/
+/
+
+CMACMA:	ASCIZ/,,/
+
+SEP:	ASCIZ/-----
+/
+
+bletch: asciz\
+WRONG SYSTEM VERSION - REPURIFY
+\
+
+MEMHOL:	ASCIZ\
+WARNING: THERE IS A HOLE IN MEMORY
+\
+
+	loc 164
+
+STACK=100		;DEFAULT TO 100 WORDS
+pdl: 	BLOCK STACK
+
+	loc 264		; not sure why removing this causes errors
+
+sysmsg: move .cf.,[-stack,,memhol+7]
+	move p,[-stack,,memhol+7]
+	.call [setz ? sixbit/OPEN/ ? [1,,1] 
+      	  [sixbit/DSK/] ? [sixbit/CDATA/] ? [sixbit/>/] ? setz [sixbit/DRAGON/]]
+	.value
+        pushj p,ritual		; assurance of purity
+        movei count,t1
+        lsh count,		; instruction patched through abstb1
+        movei pt,               ; instruction patched through abstb1
+        jrst papsav
+
+g00002:
+    skipn .prsva,(pt)
+    jrst nxloop
+    jumpge .prsva,g00002+5
+    movei t1,[feep]
+    pushj p,type
+    hllz f,.prsva
+    hrli .prsva,440700
+    movei c,t1(pt)
+loop1:
+    ildb b,.prsva
+    jumpe b,g00013
+    caie b,pdl+12
+    jrst loop1+6
+    pushj p,argot
+    jrst g00011
+    tyo b
+g00011:
+    jrst loop1
+g00013:
+    tlnn f,gotit+4
+    jrst g00014
+    tyo [40]
+    pushj p,argot
+    jrst g00013
+g00014:
+    movei t1,[crlf]
+    pushj p,type
+nxloop:
+    addi pt,.prsva
+    sojg count,g00002
+g00003:
+    movei t1,[sep]
+    pushj p,type
+papsav:
+    movei c,
+    move pt,(c)
+syfndl:
+    pushj p,syget
+    tyo .prsva
+    cain .prsva,.pf.
+    tyo [count]
+    jrst syfndl
+syget0:
+    camn pt,(c)
+    .hang
+    camn pt,(c)
+    subi pt,
+    ildb .prsva,pt
+    popj p,
+
+syget:	pushj p,syget0
+	trnn a,200		; and do special char processing
+    	popj p,			; normal char, just return
+    	cain a,%TDFS		; cursor forward one column
+    	jrst [movei a,40 ? popj p,]
+    	cain a,%TDMV0		; move cursor, followed by new vert pos, new hor pos
+    	jrst syget1
+    	caile a,%TDMV1		; dummy command created by TYMOV
+    	jrst syget
+
+syget1:  pushj p,syget0		; gobble cursor motion
+    	 pushj p,syget0		; assuming intr level has processed
+    	 trze a,200
+    	 jrst syget1
+    	 caie a,
+      	 skipa a,[40]		; otherwise is probably space
+	 movei a,15		; now change cursor motion to crlf
```

Comment:
Maybe ^M here?

Reply:
I was being faithful to the code in SYSMSG.  It uses skipa a,[40]

Reply:
And SYSMSG uses movei a,15 too.

Reply:
Ok, anything in SYSMSG overrides my suggestions.